[16.0][MIG] mail_tracking: Migration to 16.0

maq-adhoc opened the pull request:
I restart the mail tracking migration, taking changes from PR https://github.com/OCA/social/pull/1029 and adding my own changes. 
I restarted to be able to do it with the odoo-module-migrate tool and the commits are clean and include PRs #1053 ,#1056 ,#1057.  
I also modified the way we inherit MessagingInitializer so that it does not conflict with other modules, e.g. Voip.

antespi committed:
[8.0][ADD] mail_tracking addon (#67)

* [ADD] mail_tracking addon

* Add description icon

* Fixes remarked

* Fix Travis error

* Remarks fixed

antespi committed:
[8.0][IMP][mail_tracking] Speed installation time and discard concurrent events (#82)

[IMP] mail_tracking: Speed installation time, discard concurrent events and other fixes

antespi committed:
[FIX] Update email_score when email changes

antespi committed:
[MIG] mail_tracking

* Improve tests
* Show trackings even if partner removed
* Disable CSRF protection to webhooks controllers

pedrobaeza committed:
[FIX] mail_tracking: Typo

pedrobaeza committed:
[MIG] Make modules uninstallable

antespi committed:
[IMP] mail_tracking: Allow to define a custom mail_tracking base url

antespi committed:
[IMP] mail_tracking performance and bounce process (#103)

bouvyd committed:
[MIG] mail_tracking: Migrated to 10.0

antespi committed:
Add access to portal/public user to his own email trackings

antespi committed:
[FIX] Sender label is repeated in search fields

antespi committed:
[9.0][FIX][mail_tracking] Control exception when mail_message was removed (#132)

antespi committed:
[8.0][FIX][mail_tracking] Use event recipient address to find partners and contacts to bounce (#133)

jjscarafia committed:
[FIX][9.0] partner buttons view on enterprise

carlosdauden committed:
[9.0][IMP] mail_tracking: Improve view with error details

pedrobaeza committed:
[FIX] mail_tracking: Obtain outgoing server with sudo

As regular users can't access this object.

chienandalu committed:
[IMP] Shut mock exceptions up

antespi committed:
[FIX] mail_tracking: bugfixes (#199)

* [FIX] Only one data-odoo-tracking-email tag in each email
* [FIX] Get status even in multicompany instances

eLBati committed:
IMP mail_tracking performances

On our server,
queries based on "mail_tracking_event"."tracking_email_id" improved from 501,924 ms to 1,840 ms
queries based on "mail_tracking_email"."mail_message_id" improved from 167,436 ms to 3,223 ms

The last ones are run several times when a thread has many messages

chienandalu committed:
[MIG] mail_tracking: Migration to 11.0

ecino committed:
[FIX] mail_tracking js could break other modules using chat_manager

didierdonze committed:
Translated using Weblate (French) [ci skip]

Currently translated at 100.0% (82 of 82 strings)

Translation: social-11.0/social-11.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-11-0/social-11-0-mail_tracking/fr/

chienandalu committed:
[IMP] mail_tracking: email score performance (#299)

bitroniq committed:
Fix README.rst images not visible in GitHub and Odoo

Closes #303

pedrobaeza committed:
[IMP] mail_tracking: Put button box at the end of the view

For giving more priority to other buttons like the invoices one.

eduardgm committed:
Translated using Weblate (Catalan)

Currently translated at 96.3% (79 of 82 strings)

Translation: social-11.0/social-11.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-11-0/social-11-0-mail_tracking/ca/

ernestotejeda committed:
[MIG] mail_tracking: Migration to 12.0

pedrobaeza committed:
[FIX] mail_tracking: Don't call write inside other write

Or infinite recursions will happen on other `write` overwrites, like the one that happens
on `mass_mailing_partner`.

Tardo committed:
[IMP] mail_tracking: Cc

Tardo committed:
[MIG] mail_tracking: Adapt Cc to 12.0

chienandalu committed:
[IMP] mail_tracking: mail.thread filter for tracking errors

- Any model inheriting from mail.thread will have a filter available to
obtain records with errors in their messages trackings.
- The messages can be marked as done to avoid false positives when the
issues are solved.

OCA-git-bot committed:
mail_tracking 12.0.1.1.0

Tardo committed:
[FIX] mail_tracking: Repeated suggested cc emails

EgorSummer committed:
[FIX] Bug {TypeError: dict.message.hasEmailCc is not a function} during the creation of a new document.

OCA-git-bot committed:
mail_tracking 12.0.1.1.1

Tardo committed:
[IMP] mail_tracking: Adapt test to set (not order preserved)

OCA-git-bot committed:
mail_tracking 12.0.1.1.2

Tardo committed:
[IMP] mail_tracking: Cc name & check if mail is Cc

pedrobaeza committed:
[FIX] mail_tracking: Show correctly blank image

Bad P2 > P3 conversion

eduardgm committed:
Translated using Weblate (Spanish)

Currently translated at 33.3% (29 of 87 strings)

Translation: social-12.0/social-12.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-12-0/social-12-0-mail_tracking/es/

Tardo committed:
[IMP] mail_tracking: Convert 'partner_trackings' to dict

Tardo committed:
[IMP] mail_tracking: translations

Tardo committed:
[IMP] mail_tracking: Failed Messages to 12.0

OCA-git-bot committed:
mail_tracking 12.0.2.0.0

Tardo committed:
[FIX] mail_tracking: Refresh thread widget when sets a message as reviewed

Previous this commit, failed messages widget data wasn't updated properly when you
set as reviewed/retry the last message displayed.

This commit resolve the problem and improves the "set as reviewed" action to avoid
multi-user inconsistencies (don't toggle the value).

OCA-git-bot committed:
mail_tracking 12.0.2.0.1

Tardo committed:
[MIG] mail_tracking: Migration to 13.0

Tardo committed:
[IMP] mail_tracking: black, isort

Tardo committed:
[FIX] mail_tracking: Changed injection method for 'failed_message_ids' field

Before this commit, some methods with views crashes trying to
process 'failed_message_ids' field.
After this commit the 'failed_message_ids' field description
is properly generated.

oca-travis committed:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update mail_tracking.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-13.0/social-13.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_tracking/

OCA-git-bot committed:
mail_tracking 13.0.1.0.1

Tardo committed:
[IMP] mail_tracking: Store To recipients and omit aliases

Tardo committed:
[IMP] mail_tracking: Test 'to' and named emails

OCA-git-bot committed:
[UPD] README.rst

ruuter committed:
[FIX] mail_tracking: do not warn not found if already opened

The state will be 'opened' after first time email is opened. Each
successive open triggered warning, because state was already 'opened'
and thus did not match domain state in (sent, delivered).

Forward-port #d9ad3464

OCA-git-bot committed:
mail_tracking 13.0.1.0.2

Tardo committed:
[IMP] mail_tracking: Ignore catchall

oca-travis committed:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
mail_tracking 13.0.1.0.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-13.0/social-13.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_tracking/

oca-travis committed:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
mail_tracking 13.0.1.0.4

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-13.0/social-13.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_tracking/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-13.0/social-13.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_tracking/

Tardo committed:
[IMP] mail_tracking: Discuss failed messages: Button and domain

wpichler committed:
[FIX] re.search will return None when nothing found. So we have to check for this

oca-travis committed:
[UPD] Update mail_tracking.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-13.0/social-13.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_tracking/

nmadhoc committed:
[FIX] mail_tracking
When a user try to resend message gets an error stating that it's only allowed for users who belong to the Administration/Settings group

OCA-git-bot committed:
mail_tracking 13.0.1.0.6

Commit:
[IMP] pre-commit run -a

Tardo committed:
[IMP] Apply pre-commit changes: Resolve conflicts

Commit:
[FIX] pre-commit run -a

Tardo committed:
[IMP] mail_tracking: Discuss failed messages: Only show confirmation if needed

oca-travis committed:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
mail_tracking 13.0.1.0.7

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-13.0/social-13.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-13-0/social-13-0-mail_tracking/

Tardo committed:
[FIX] mail_tracking: Store cc

OCA-git-bot committed:
mail_tracking 13.0.1.0.8

AreebSiddiqiIR committed:
[IMP] mail_tracking: black, isort, prettier

AreebSiddiqiIR committed:
[MIG] mail_tracking: Migration to 14.0

SimoneVagile committed:
[14.0][MIG] mail_tracking

angelJuaGar committed:
[FIX] mail_tracking: Modification in unitest.

oca-travis committed:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[FIX] mail_tracking: reduce spammy score

Fix https://github.com/OCA/social/issues/701, following idea from https://kb.mailwizz.com/articles/low-score-in-spamassassin-because-of-the-rand_mktg_header-rule/ to avoid matching the regexp explained in https://bz.apache.org/SpamAssassin/show_bug.cgi?id=7888#c5

@Tecnativa TT29674

angelJuaGar committed:
[FIX] mail_tracking: Added restriction on email search.

A filter_domain was added in the sender and recipient fields to be able to handle a more specific filter when using mail tracking.

OCA-git-bot committed:
mail_tracking 14.0.1.1.0

JorgeQuinteros committed:
[FIX] Change operator to 'ilike'

OCA-git-bot committed:
mail_tracking 14.0.1.1.1

oca-travis committed:
[UPD] Update mail_tracking.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-14.0/social-14.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-14-0/social-14-0-mail_tracking/

Commit:
[FIX] mail_tracking: process sent events

Up until now, the `sent` event type was never processed, probably because it "made no sense", as an unsent email would never trigger an event.

However, it makes sense to process it because you may have a local relay that transmits mails over to the mail provider. In those circumstances, you should have 2 "sent" events (one from the relay and another one from the provider).

Also marked some useless parts of code for removal.

@Tecnativa TT32365

oca-travis committed:
[UPD] Update mail_tracking.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-14.0/social-14.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-14-0/social-14-0-mail_tracking/

houssine78 committed:
[PORT] porting js code to comply with OWL

houssine78 committed:
[PEP8] pre-commit

oca-travis committed:
[UPD] Update mail_tracking.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-14.0/social-14.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-14-0/social-14-0-mail_tracking/

oca-travis committed:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
mail_tracking 14.0.2.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-14.0/social-14.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-14-0/social-14-0-mail_tracking/

fernandahf committed:
[FIX] mail_tracking: use recipient_address to search emails

This commits is using the field recipient_address in filter of search view
of mail.tracking.email in order to use the same field used in computed
method:

https://github.com/OCA/social/blob/147eda96a56c199602c619b5bac3a9911605e918/mail_tracking/models/res_partner.py#L27

fernandahf committed:
[FIX] mail_tracking: change order of inheritance and use return

Currently when a invoice view is reload in browser or through native
redirections, an error is raised:

```
Traceback:
TypeError: Cannot read properties of undefined (reading 'id')
at Function.convertData [as _super] (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:5768:118)
at Function.convertData (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:6651:335)
at Function.Class. (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:6478:107)
at Function.convertData (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:7779:403)
at Function.Class. (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:6478:107)
at Function.convertData (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:7818:353)
at Function.Class. (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:6478:107)
at Function.convertData (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:12197:435)
at Function.Class. (https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:6478:107)
at https://t50084-053-islamicrelief140.dev.irc.deployv.com/web/content/1819726-9a8777a/web.assets_backend.js:5790:359
```

That is because an inheritance wrong, which breaks the flow of start of model
`messaging` and causes that messages execute before the init of var `messaging`.

oca-travis committed:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
mail_tracking 14.0.2.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-14.0/social-14.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-14-0/social-14-0-mail_tracking/

OCA-git-bot committed:
mail_tracking 14.0.2.0.2

sugapablo committed:
[FIX] mail_tracking: error when filtering by "Sent"

JasminSForgeFlow committed:
[IMP] mail_tracking: black, isort, prettier

JasminSForgeFlow committed:
[MIG] mail_tracking: Migration to 15.0

Commit:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
mail_tracking 15.0.1.0.1

Commit:
[IMP] mail_tracking: remove legacy hooks

Commit:
Fixed Deprecated Warning

OCA-git-bot committed:
mail_tracking 15.0.1.0.2

Commit:
[UPD] Update mail_tracking.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-15.0/social-15.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-15-0/social-15-0-mail_tracking/

chienandalu committed:
[FIX] mail_tracking: permissions

Refine which mail tracking record is allowed to see. We can rely on the
ACLs of the related records so we minimize inconvenient message subject
leaks.

A regular user can read these mail.tracking.email records:

- Those with a linked mail.message that the user can read
- Those with a linked mail.mail that the user can read
- Those with no message/mail link but a linked partner that the user can
read.
- Those with no linked records.

TT31399

chienandalu committed:
[IMP] mail_tracking_mailgun: mass mailing manual tracking

Mass mailing are tracked from mail.trace as the don't store a message in
the db. In order to gather the message_id and be able to do manual
checks to mailgun, that's the table where we should get the message id.

TT40816

chienandalu committed:
[FIX] mail_tracking: mixed classes

In migration #873 the classes DiscussController and MailController which
are the result of an upstream functionality split are merge altogether
in this module override causing undesired side effects in other modules
trying to extend those (i.e.: portal)

TT41432

Commit:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
mail_tracking 15.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-15.0/social-15.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-15-0/social-15-0-mail_tracking/

chienandalu committed:
[FIX] mail_tracking: js linter warnings

Commit:
[UPD] Update mail_tracking.pot

OCA-git-bot committed:
mail_tracking 15.0.2.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: social-15.0/social-15.0-mail_tracking
Translate-URL: https://translation.odoo-community.org/projects/social-15-0/social-15-0-mail_tracking/

jonasbuchholz committed:
use keyword arguments in build_email to comply with portal tests

OCA-git-bot committed:
mail_tracking 15.0.2.0.1

OCA-git-bot committed:
mail_tracking 15.0.2.0.2

filoquin committed:
[MIG] mail_tracking: Migration to 16.0

filoquin committed:
[IMP] mail_tracking: pre-commit execution

filoquin committed:
[MIG] mail_tracking: Migration to 16.0

JrAdhoc committed:
[FIX] mail_tracking: tuple index out of range fix

filoquin committed:
[IMP] Add tracking_delta for use as owl for each unique  key

pedrobaeza commented:
There's already #1029, so please review that one. Closing for now.

pedrobaeza commented:
Sorry, I have seen the comments, so this one may be the correct one. Please check pre-commit.

pedrobaeza commented:
/ocabot migration mail_tracking

remi-filament commented on mail_tracking/static/src/xml/failed_message/discuss.xml:
```diff
@@ -0,0 +1,65 @@
+            expr="//t[@t-elif='thread.mailbox === messaging.history']"
+            position="after"
+        >
+            <t t-elif="thread === messaging.failedmsg">
+                <div class="o_ThreadIcon_mailboxModeration fa fa-exclamation" />
```
```suggestion
            <t t-elif="thread.mailbox === messaging.failedmsg">
                <div class="o_ThreadIcon_mailboxModeration fa fa-fw fa-exclamation" />
```
The failedmsg mailbox is showing without icon without the above change

remi-filament commented on mail_tracking/static/src/xml/failed_message/discuss.xml:
```diff
@@ -0,0 +1,65 @@
+                <div class="o_MessageList_emptyTitle o-neutral-face-icon">
+                    No failed messages
+                </div>
+                Failed messages will be appeared here.
```
```suggestion
                Failed messages will be listed here.
```

remi-filament commented on mail_tracking/readme/USAGE.rst:
```diff
@@ -0,0 +1,75 @@
+
+* Discuss
+
+  .. image:: ../static/img/failed_message_discuss.png
```
Image is quite old and could be replaced by a more up to date UI capture

maq-adhoc replied:
thanks @remi-filament the errors in Failedmsg had been detected and I am already working to fix them. I take note of the rest of the comments to verify and correct them.

nguyenminhchien replied:
@maq-adhoc i just created this [Pull request](https://github.com/adhoc-dev/social/pull/1) to fix:
- Failedmsg on Discuss app
- Action in Discuss App to "Set as reviewed" or "Retry" have no effect

Please review
Thanks,

remi-filament reviewed (changes requested):
Hi @maq-adhoc, Thank you for this port !
I am not sure how to test this one extensively, so far I have spotted a problem of display on Discuss page.

Some other issues I found (things that I have verified are working in v15) : 
-  I do not see any message in Failedmsg on Discuss app, although I guess there should be the demo message created for that purpose no ?
- Failed sent messages filter is not added in search views
- Action in Discuss App to "Set as reviewed" or "Retry" have no effect

kmagusiak commented on mail_tracking/models/mail_tracking_email.py:
```diff
@@ -0,0 +1,472 @@
+    time = fields.Datetime(readonly=True, index=True)
+    date = fields.Date(readonly=True, compute="_compute_date", store=True)
+    mail_message_id = fields.Many2one(
+        string="Message", comodel_name="mail.message", readonly=True, index=True
```
```suggestion
        string="Message (mail)", comodel_name="mail.message", readonly=True, index=True
```
Name conflicts with `message_id` field.

nguyenminhchien commented on mail_tracking/models/ir_mail_server.py:
```diff
@@ -0,0 +1,134 @@
+                smtp_password=smtp_password,
+                smtp_encryption=smtp_encryption,
+                smtp_ssl_certificate=smtp_ssl_certificate,
+                smtp_ssl_private_key=smtp_ssl_certificate,
```
@maq-adhoc it should be `smtp_ssl_private_key=smtp_ssl_private_key` instead of `smtp_ssl_private_key=smtp_ssl_certificate`

nguyenminhchien commented on mail_tracking/models/mail_thread.py:
```diff
@@ -0,0 +1,136 @@
+                )
+
+    @api.model
+    def _fields_view_get(
```
@maq-adhoc  Method `_fields_view_get` is deprecated, use `_get_view` instead

nguyenminhchien reviewed (changes requested):
I added 2 comments, please check them. Thank you.

TDu committed:
Fix mail tracking bounce notification not logged

This is to resolve the following issue:

* https://github.com/OCA/social/issues/1121

TDu commented:
@maq-adhoc  Could you include this fix https://github.com/OCA/social/pull/1139 that has been merged into 14. Thank you

chienandalu committed:
[FIX] mail_tracking: thread permissions

If a user tries to read a thread on a record and one message is from a
partner on which he has no permissions, there will be an exception as
this method tries to fetch info from such partner.

TT43075

pedrobaeza commented:
Please cherry-pick #1141

chienandalu committed:
[FIX] mail_tracking: allowed trackings compute

In the forward port of the permission check, a column was dropped from
the tuple, but the following code wasn't adapted to the new tuple
length. For clarity sake we also name the tuple unpacking into proper
variables.

TT43453

pedrobaeza commented:
Please cherry-pick #1149

nguyenminhchien commented on mail_tracking/models/mail_thread.py:
```diff
@@ -0,0 +1,141 @@
+
+        arch, view = super()._get_view(view_id, view_type, **options)
+        if view_type in {"search", "form"}:
+            doc = etree.XML(arch)
```
I face an error when accessing the Sales menu. Seems that you don't test before pushing your changes.
![image](https://github.com/OCA/social/assets/8724793/5de383eb-509a-4d4f-80df-b0d28e7259ba)

maq-adhoc replied:
It was a work commit. I fixed that error. The module is functional for almost everything.

Not all the problems are solved, e.g. it doesn't list all the error cases in the discuss/failed msg menu. This week I think the final version will be ready.

maq-adhoc committed:
[FIX] Working commit

nguyenminhchien committed:
[FIX] Failedmsg on Discuss, Set as reviewed, Retry

maq-adhoc commented:
> I added 2 comments, please check them. Thank you.
@nguyenminhchien 

They are made in the current commit. 

https://github.com/OCA/social/pull/1091/files#diff-3bb501b5e868a62e76b5c295015ee18984cf388955d4b160442f145b39c3e694R121
https://github.com/OCA/social/pull/1091/files#diff-cbb61b601dc8e9b4a1de5d64eeabd4f63cffa4f007f7cad220787ea714a3f83eR90

nguyenminhchien commented:
> They are made in the current commit.

@maq-adhoc thanks. But, i just found an issue need to be migrated: Showing the Failedmsg on the Chatter
![image](https://github.com/OCA/social/assets/8724793/ab8a6425-4a8b-4277-9d3a-752d0f0f61a3)

maq-adhoc committed:
Merge pull request #1 from nguyenminhchien/16.0-mig-mail_tracking_1091

[FIX] Failedmsg on Discuss, Set as reviewed, Retry

maq-adhoc committed:
[FIX] fix format

nguyenminhchien reviewed (changes requested):
**Failed msg** on the Chatter is not shown

nguyenminhchien committed:
[UPD] Show Failedmsg on Chatbox

pedrobaeza commented:
/ocabot migration mail_tracking

nguyenminhchien commented:
@maq-adhoc : i created this [PR](https://github.com/adhoc-dev/social/pull/2) to show the Failed Messages on Chatbox

maq-adhoc committed:
Merge pull request #2 from nguyenminhchien/16.0-mig-mail_tracking_1091

[UPD] Show Failedmsg on Chatbox

nguyenminhchien reviewed (approved):
lgtm

patel1008b commented:
https://github.com/OCA/social/pull/1191

payen000 commented on mail_tracking/static/src/js/models/mailbox.esm.js:
```diff
@@ -0,0 +1,58 @@
+            compute() {
+                switch (this) {
+                    case this.messaging.failedmsg:
+                        return this.env._t("Failedmsg");
```
```suggestion
                        return this.env._t("Failed");
```

To maintain consistency with previous versions, I think this shoud say "Failed" instead.

payen000 commented on mail_tracking/static/src/js/failed_message/thread.esm.js:
```diff
@@ -0,0 +1,58 @@
+/** @odoo-module **/
```
This file seems to not be needed anymore, so we can delete it

payen000 commented on mail_tracking/static/src/js/failed_message/discuss.js:
```diff
@@ -0,0 +1,411 @@
+/* Copyright 2019 Alexandre Díaz
```
Is this file in use? If not, I think we should delete it before merging

payen000 reviewed (commented):
I tested this functionally and seems to be working as expected.

This will be ready to merge once the unused files are deleted and the commits are sorted out (as there are too many commits right now).

Edit:

My bad, I didn't realize the many commits were the module's commit history.

payen000 commented:
Hello @maq-adhoc,

I left some comments for you, could you check them please?

I believe the module is working as it should and will be ready to merge once we delete the unused file(s) and switch "Failedmsg" by "Failed", or is there anything else that I'm missing?

cvinh reviewed (changes requested):
Hello
Could you please forward port this PR : https://github.com/OCA/social/pull/941 ?
Thanks

payen000 commented:
Superseded by https://github.com/OCA/social/pull/1216

However, I do believe this PR was already working properly, I'm only opening the other PR due to inactivity here, as I need the module working in `v16` as soon as possible.
